### ISSUE TITLE
When creating ColorPreference programmatically defaultValue not works

### ISSUE THREAD

**Issue by matzrm** [martin-stone/hsv-alpha-color-picker-android#45]
I'm creating preference screen programmatically and I'm putting a ColorPrefrence inside it in this way:

```
ColorPreference pref= new ColorPreference(getActivity(),null);
pref.setTitle("Color");
pref.setKey("pref_color");
pref.setDefaultValue("#000000"); // or Color.WHITE
```

The problem is that until I change the color the thumbail isn't shown, because in method _onSetInitialValue()_ of ColorPreference defaultValue is loaded right but not used as argument into _setColor()_. Programmatically there'isnt a way to set the defaultColor. Isn't it?

```
@Override
protected void onSetInitialValue(boolean restorePersistedValue, Object defaultValue) {
	setColor(restorePersistedValue ? getColor() : defaultColor);
}
```

**Comment by martin-stone** [comment]
Yes I think onGetDefaultValue and onSetInitialValue may not be doing quite the right thing at the moment. Looking into it.

**Comment by martin-stone** [comment]
Fixed in 2.4.0.
